lib/constructorio.js

```diff
@@ -319,6 +319,115 @@ ConstructorIO.prototype = {
     });
   },
 
+  /**
+   * @description Adds a synonym group.
+   */
+  addSynonymGroup(params, callback) {
+    request({
+      method: 'POST',
+      url: this.makeUrl('synonym_groups'),
+      auth: this.makeAuthToken(),
+      json: params,
+    }, (error, response) => {
+      handleServerResponse(error, response, callback);
+    });
+  },
+
+  /**
+   * @description Modifies a synonym group for supplied id.
+   */
+  modifySynonymGroup(params, callback) {
+    const json = clonedeep(params);
+    const { group_id } = json;
+    delete json.group_id;
+
+    request({
+      method: 'PUT',
+      url: this.makeUrl(`synonym_groups/${group_id}`),
+      auth: this.makeAuthToken(),
+      json: params,
+    }, (error, response) => {
+      handleServerResponse(error, response, callback);
+    });
+  },
+
+  /**
+   * @description Retrieves all synonym groups optionally filtered by phrase
+   */
+  getSynonymGroups(params, callback) {
+    const { num_results_per_page, phrase, page } = params;
+    const qsParams = {};
+
+    if (num_results_per_page) {
```

Comment:
do we need these if checks here?

Reply:
Mostly just did this to follow the convention of the methods above (see `getItem` on line 182).